common/main/java/com/couchbase/lite/AbstractQuery.java

```diff
@@ -83,13 +83,23 @@
      * Set parameters should copy the given parameters. Set a new parameter will
      * also re-execute the query if there is at least one listener listening for
      * changes.
-     * METHOD NEEDS TO BE REWRITTEN with core c4query_setParameters
+     *
+     * @throws IllegalStateException    on failure to create the query (e.g., database closed)
+     * @throws IllegalArgumentException on failure to encode the parameters (e.g., parameter value not supported)
      */
     @Override
     public void setParameters(@Nullable Parameters parameters) {
-        synchronized (lock) {
-            this.parameters = (parameters == null) ? null : parameters.readonlyCopy();
+        synchronized (lock) { this.parameters = (parameters == null) ? null : parameters.readonlyCopy(); }
+        try {
```

Comment:
You don't need nested try blocks... you just need multiple catch blocks

Also, I think it would be better to wrap the smallest possible amount of code, in the try.